build.gradle

```diff
@@ -93,48 +123,49 @@ allprojects {
     // Default to the apache license
     project.ext.licenseName = 'The Apache Software License, Version 2.0'
     project.ext.licenseUrl = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-    plugins.withType(ShadowPlugin).whenPluginAdded {
```

Comment:
We can't remove this portion its referenced here: https://github.com/opensearch-project/job-scheduler/blob/main/scripts/build.sh#L79

Adding this back in.

Reply:
I moved this down below as a second maven repo that the artifacts are published to. I'm testing locally with an older commit before the gradle update and now to compare what gets published.

Reply:
Everytime I test with an older version of gradle before the gradle upgrade I get this error: 

```
./gradlew publishAllPublicationsToStagingRepository

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'opensearch-job-scheduler'.
> java.util.concurrent.ExecutionException: org.gradle.api.GradleException: Failed to create Jar file /Users/cwperx/.gradle/caches/jars-9/94ac7eb5f0b504d7e3b47bc2eec907cc/jackson-core-2.15.0.jar.
```

Clearing the cache does not help. Remove the lockfiles also does not help. I also tried rebooting my computer and receive the same error.

Reply:
This is the output of `./gradlew publishAllPublicationsToStagingRepository` on this branch.

```
➜  job-scheduler git:(artifact) ✗ ./gradlew publishAllPublicationsToStagingRepository
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 8.0.2
  OS Info               : Mac OS X 13.3.1 (x86_64)
  JDK Version           : 11 (Amazon Corretto JDK)
  JAVA_HOME             : /Users/cwperx/.sdkman/candidates/java/11.0.17-amzn
  Random Testing Seed   : 30E4AD4FE4FC7F2E
  In FIPS 140 mode      : false
=======================================

> Task :compileJava
Note: /Users/cwperx/Projects/opensearch/job-scheduler/src/main/java/org/opensearch/jobscheduler/JobSchedulerPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :publishPluginJarPublicationToStagingRepository
Multiple publications with coordinates 'org.opensearch:opensearch-job-scheduler:3.0.0.0-SNAPSHOT' are published to repository 'Staging'. The publications 'nebula' in root project 'opensearch-job-scheduler' and 'pluginJar' in root project 'opensearch-job-scheduler' will overwrite each other!

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.0.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 24s
24 actionable tasks: 24 executed
➜  job-scheduler git:(artifact) ✗ tree build/local-staging-repo
build/local-staging-repo
└── org
    └── opensearch
        ├── opensearch-job-scheduler
        │   ├── 3.0.0.0-SNAPSHOT
        │   │   ├── maven-metadata.xml
        │   │   ├── maven-metadata.xml.md5
        │   │   ├── maven-metadata.xml.sha1
        │   │   ├── maven-metadata.xml.sha256
        │   │   ├── maven-metadata.xml.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-javadoc.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-javadoc.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-javadoc.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-javadoc.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-javadoc.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-sources.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-sources.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-sources.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-sources.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1-sources.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.module
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.module.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.module.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.module.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.module.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-javadoc.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-javadoc.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-javadoc.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-javadoc.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-javadoc.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-sources.jar
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-sources.jar.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-sources.jar.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-sources.jar.sha256
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2-sources.jar.sha512
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2.pom
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2.pom.md5
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2.pom.sha1
        │   │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-2.pom.sha256
        │   │   └── opensearch-job-scheduler-3.0.0.0-20230504.204742-2.pom.sha512
        │   ├── maven-metadata.xml
        │   ├── maven-metadata.xml.md5
        │   ├── maven-metadata.xml.sha1
        │   ├── maven-metadata.xml.sha256
        │   └── maven-metadata.xml.sha512
        ├── opensearch-job-scheduler-spi
        │   ├── 3.0.0.0-SNAPSHOT
        │   │   ├── maven-metadata.xml
        │   │   ├── maven-metadata.xml.md5
        │   │   ├── maven-metadata.xml.sha1
        │   │   ├── maven-metadata.xml.sha256
        │   │   ├── maven-metadata.xml.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-javadoc.jar
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-javadoc.jar.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-javadoc.jar.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-javadoc.jar.sha256
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-javadoc.jar.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-sources.jar
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-sources.jar.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-sources.jar.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-sources.jar.sha256
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1-sources.jar.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.jar
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.jar.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.jar.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.jar.sha256
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.jar.sha512
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.pom
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.pom.md5
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.pom.sha1
        │   │   ├── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.pom.sha256
        │   │   └── opensearch-job-scheduler-spi-3.0.0.0-20230504.204742-1.pom.sha512
        │   ├── maven-metadata.xml
        │   ├── maven-metadata.xml.md5
        │   ├── maven-metadata.xml.sha1
        │   ├── maven-metadata.xml.sha256
        │   └── maven-metadata.xml.sha512
        └── plugin
            └── opensearch-job-scheduler
                ├── 3.0.0.0-SNAPSHOT
                │   ├── maven-metadata.xml
                │   ├── maven-metadata.xml.md5
                │   ├── maven-metadata.xml.sha1
                │   ├── maven-metadata.xml.sha256
                │   ├── maven-metadata.xml.sha512
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.md5
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.sha1
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.sha256
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.pom.sha512
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.zip
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.zip.md5
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.zip.sha1
                │   ├── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.zip.sha256
                │   └── opensearch-job-scheduler-3.0.0.0-20230504.204742-1.zip.sha512
                ├── maven-metadata.xml
                ├── maven-metadata.xml.md5
                ├── maven-metadata.xml.sha1
                ├── maven-metadata.xml.sha256
                └── maven-metadata.xml.sha512
```